qemu/tests/image_creation_lock_release.py

```diff
@@ -0,0 +1,25 @@
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    Image creation locking should be properly released.
+    1. Create raw image and close it after 0.5 sec.
+    2. Check there is no error of the CML.
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    lock_test = params["images"]
+    root_dir = data_dir.get_data_dir()
+    test_image = QemuImg(params.object_params(lock_test), root_dir, lock_test)
+    test_filename = test_image.image_filename
+    try:
+        process.run("qemu-img create -f raw -o preallocation=full %s 1G & "
+                    "sleep 0.5;qemu-io -c info -c close -r %s"
+                    % (test_filename, test_filename), shell=True)
```

Comment:
I thought you should try this command `qemu-img create -f raw -o preallocation=full %s 1G & sleep 0.5;qemu-io -c info -c close -r`  if is right.

My test:
```
>>> from avocado.utils import process
>>> process.run("qemu-img create -f raw -o preallocation=fullxxdxxx aa.raw 1G & sleep 0.5; uname -r ", shell=True)
<avocado.utils.process.CmdResult object at 0x7f975c0d1f28>
```
Not raise` process.CmdError`

So I think if here hit the bug like:
```
# qemu-img create -f raw -o preallocation=full foo.img 1G &  sleep 0.5; qemu-io -c info -c close -r foo.img
[1] 19485
Formatting 'foo.img', fmt=raw size=1073741824 preallocation=full
can't open device foo.img: Failed to get "consistent read" lock
Is another process using the image?
```
You couldn't found it since your case doesn't check it. so this case will pass.

And I thought the return status of the above command `qemu-img create -f raw -o preallocation=full foo.img 1G &  sleep 0.5; qemu-io -c info -c close -r foo.img` is `0` under the problem qemu version.

like:
```
[root@hp-dl388g8-16 avocado]# qemu-img create -f raw -o preallocation=fullxxdxxx /home/aa.raw 1G & sleep 0.5; uname -r
[1] 8497
Formatting '/home/aa.raw', fmt=raw size=1073741824 preallocation=fullxxdxxx
qemu-img: /home/aa.raw: invalid parameter value: fullxxdxxx
[1]+  Exit 1                  qemu-img create -f raw -o preallocation=fullxxdxxx /home/aa.raw 1G
4.18.0-147.0.3.el8_1.x86_64
[root@hp-dl388g8-16 avocado]# echo $?
0
[root@hp-dl388g8-16 avocado]# 
```

Reply:
For your first question, I tried the case in the qemu version - qemu-kvm-rhev-2.12.0-7.el7(The original production bug occur in this package.). It failed as expected.
`21:15:48 ERROR|
21:15:48 ERROR| Reproduced traceback from: /home/kar/workspace/avocado/avocado/core/test.py:844
21:15:48 ERROR| Traceback (most recent call last):
21:15:48 ERROR|   File "/home/kar/workspace/avocado-vt/avocado_vt/test.py", line 301, in runTest
21:15:48 ERROR|     raise self.__status  # pylint: disable=E0702
21:15:48 ERROR| TestFail: Test failed: Command 'qemu-img create -f raw -o preallocation=full /home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw 1G & sleep 0.5;qemu-io -c info -c close -r /home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw' failed.
21:15:48 ERROR| stdout: "Formatting '/home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw', fmt=raw size=1073741824 preallocation=full\n"
21:15:48 ERROR| stderr: 'can\'t open device /home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw: Failed to get "consistent read" lock\nIs another process using the image?\n'
21:15:48 ERROR| additional_info: None
21:15:48 ERROR|
21:15:48 DEBUG| Local variables:
21:15:48 DEBUG|  -> self <class 'avocado_vt.test.VirtTest'>: 1-Host_RHEL.m7.u8.raw.virtio_scsi.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.image_creation_lock_release
21:15:48 DEBUG| DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
21:15:48 DEBUG| DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
21:15:48 DEBUG| DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
21:15:48 ERROR| FAIL 1-Host_RHEL.m7.u8.raw.virtio_scsi.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.image_creation_lock_release -> TestFail: Test failed: Command 'qemu-img create -f raw -o preallocation=full /home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw 1G & sleep 0.5;qemu-io -c info -c close -r /home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw' failed.
stdout: "Formatting '/home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw', fmt=raw size=1073741824 preallocation=full\n"
stderr: 'can\'t open device /home/kvm_autotest_root/images/rhel77-64-virtio-scsi.raw: Failed to get "consistent read" lock\nIs another process using the image?\n'
`
I think you did not catch the `process.CmdError` because of the wrong CML, NOT because of the CML fails.


As for the second question, the second CML you input is `uname -r` which is always right, so whenever the exit code is `0`. 
While in our production bug, the second CML fails because of image is locking.
Your input a wrong first, but catch the result of second CML,that has no meaning to this case/bug.

Reply:
OK